gramps/plugins/db/dbapi/dbapi.py

```diff
@@ -1183,3 +1185,17 @@ def _sql_cast_list(self, values):
         in the appropriate type.
         """
         return [v if not isinstance(v, bool) else int(v) for v in values]
+
+    def select(self, table_name, where, order_by=[], env=None):
+        """
+        Select json_data from table_name where (string).
+        """
+        evaluator = Evaluator(
+            table_name, 'json_data->>"$.{attr}"', env if env is not None else globals()
+        )
+        where_expr = str(evaluator.convert(where))
+        order_by_expr = evaluator.get_order_by(order_by)
+        self.dbapi.execute(
+            f"SELECT json_data from {table_name} WHERE {where_expr}{order_by_expr};"
```

Comment:
If you could make the where_expr optional, if would give a convenient way to retrieve all records from a table
`DBAPI.select("person")` would return all records from person

Discussion originated [here](https://github.com/gramps-project/gramps/pull/1829#issuecomment-2562769708)

Reply:
Indeed. It wasn't clear from this interface, but where="" makes it optional. I changed the interface to make this clear.